xwiki-platform-core/xwiki-platform-image/xwiki-platform-image-lightbox/xwiki-platform-image-lightbox-webjar/src/main/webjar/lightbox.js

```diff
@@ -210,20 +217,32 @@ define('xwiki-lightbox', [
       // Remember the index of the image to show first.
       $('.openLightbox').data('index', [...lightboxImages].indexOf(img));
     }).on('inserted.bs.popover', function() {
-      $('.popover .imageDownload').attr('href', this.src);
-      $('.popover .imageDownload').attr('download', getImageName(this.src));
-      var imageId = getImageId(this);
+      var img = $("#imagePopoverContainer").data('target');
+      $('.popover .imageDownload').attr('href', img.src);
+      $('.popover .imageDownload').attr('download', getImageName(img.src));
+      var imageId = getImageId(img);
       if (imageId) {
         $('.popover .permalink').attr('href', '#' + imageId);
         $('.popover .permalink').removeClass('hidden');
         $('.popover .copyImageId').attr('data-imageId', imageId);
         $('.popover .copyImageId').parent().removeClass('hidden');
       }
-    }).on('mouseenter', function() {
-      clearTimeout(timeout);
+    });
+    lightboxImages.on('mouseenter', function(e) {
+      clearTimeout(hideTimeout);
+      popoverContainer.data('target', e.target);
       $(this).popover('show');
```

Comment:
I find it strange that you call the ``.popover()`` function on 3 different things:

* here you call it on the image (``this`` is an image here)
* above you call it on ``$('#imagePopoverContainer')``
* and you also call it on ``$('#imagePopoverContainer .popover')``

Are you sure it works fine in all these cases? Are all these referring to the same popover? I'd be surprised if that is the case.

Reply:
Also, you're showing the popover directly (immediately) on mouse enter and then you hide it directly (immediately) on mouse move. Normally a mouse enter event is always followed by a mouse move (it's really hard to move the mouse on top of some element without moving it over the element). Isn't it point-less to show and then immediately hide the popover?

Reply:
Indeed, there is no need to show it on mouseenter, which was anyway misused.
But `$('#imagePopoverContainer .popover')` and `$('#imagePopoverContainer')` are pointing to the same popover, since you can use the element which triggered the popover and the popover itself for referencing the same popover.